jobs/job_utils.py

```diff
@@ -55,27 +55,70 @@ def clone_model(model, **new_values):
     return cls(id=model_id, **props)
 
 
-def get_model_kind(item):
-    """Returns the "kind", a globally unique identifier, of the given item.
+def get_model_kind(model):
+    """Returns the "kind" of the given model.
 
     NOTE: A model's kind is usually, but not always, the same as a model's class
-    name. This function will always return the correct value for "kind", even if
-    it is different from the class's name.
+    name. This function will always return the correct value for "kind", even
+    when it is different from the class's name.
 
     Args:
-        item: base_models.Model|cloud_datastore_types.Entity. The item to
+        model: base_models.Model|cloud_datastore_types.Entity. The model to
             inspect.
 
     Returns:
-        str. The item's kind.
+        bytes. The model's kind.
 
     Raises:
         TypeError. When the argument is not a model.
     """
-    if isinstance(item, base_models.BaseModel) or (
-            isinstance(item, type) and issubclass(item, base_models.BaseModel)):
-        return item._get_kind() # pylint: disable=protected-access
-    elif isinstance(item, cloud_datastore_types.Entity):
-        return item.kind
-    else:
-        raise TypeError('%r is not a model type' % type(item).__name__)
+    if isinstance(model, base_models.BaseModel) or (
+            isinstance(model, type) and
+            issubclass(model, base_models.BaseModel)):
+        return model._get_kind() # pylint: disable=protected-access
+    if isinstance(model, cloud_datastore_types.Entity):
+        return model.kind
+    raise TypeError('%r is not a model type or instance' % model)
+
+
+def get_model_property(model, property_name):
+    """Returns the given property from a model.
+
+    Args:
+        model: base_models.Model|cloud_datastore_types.Entity. The model to
+            inspect.
+        property_name: str. The name of the property to extract.
+
+    Returns:
+        *. The property's value.
+
+    Raises:
+        TypeError. When the argument is not a model.
+    """
+    if property_name == 'id':
+        return get_model_id(model)
+    if isinstance(model, base_models.BaseModel):
```

Comment:
After the Python 3 migration, will both this and the next case still exist simultaneously, or will one or the other of these branches be deleted? If the latter, maybe track with an issue, here and elsewhere.

Reply:
I haven't decided yet, I think we need more experience with the Pipelines first. There's no action to do until then, so I don't think a TODO is appropriate yet.